src/peer.ts

```diff
@@ -0,0 +1,60 @@
+import net from "net";
+import {EventEmitter} from "events";
+
+export interface PeerOpts {
+  ip: string;
+  port: number;
+}
+
+export default class Peer extends EventEmitter {
+  public ip: string;
+  public port?: number;
+  public connection: net.Socket;
+
+  /**
+   * Constructor
+   * @param {PeerOpts} opts
+   */
+  public constructor(opts: PeerOpts) {
+    super();
+    this.ip = opts.ip;
+    this.port = opts.port || 9000;
+  }
+
+  /**
+   * Establishes a connection to the peer.
+   * @returns {boolean}
+   */
+  public connect = (): Promise<boolean> => {
+    // Attempt to connect to peer, if connection refused remove the peer from bootnodes.
+    const that = this;
```

Comment:
I think you can use `this` here and don't need `that`.

Reply:
I'll run a test -- as @mpetrunic pointed out I feel there are scoping issues -- i'll write a quick test for that

Reply:
quickly speaking I found this: 

javascript, promises, how to access variable this inside a then scope
http://stackoverflow.com/questions/32547735/ddg#32547793

I'll write up a gist though

Reply:
If I remember correctly, you lose scope when you use arrow function, that's why we use `function(){}` instead of `() => {}` in mocha if we are going to use something like `this.timeout()`

Reply:
```
> $ cat test.js

class Test {
 constructor() {
   this.a = "A";
 }

  start() {
    console.log('outer this', this.a);
    let that = this;
    return new Promise(function (reject,resolve) {
      setTimeout(function() {
        console.log('inner this', this.a);
	console.log('inner that', that.a);
        resolve('foo');
      }, 300);
    })
  }
}

const t = new Test();
t.start();

> $ node test.js

outer this A
inner this undefined
inner that A
(node:97385) UnhandledPromiseRejectionWarning: foo
(node:97385) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:97385) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```